% !TEX encoding = UTF-8 Unicode
\documentclass[11pt]{article}

\usepackage[letterpaper, margin=1.5cm]{geometry}
\usepackage{graphicx}
\usepackage[english]{babel}
\usepackage[utf8x]{inputenc}
\usepackage{amsmath,amsfonts,amssymb}
\usepackage{hyperref}
\usepackage[square,numbers,sort&compress]{natbib}
\newcommand{\TODO}[1]{\begingroup\color{red}#1\endgroup}
\newcommand{\PFS}[1]{\begingroup\color{blue}#1\endgroup}
\newcommand{\CAVH}[1]{\begingroup\color{green}#1\endgroup}
\usepackage{color}
\usepackage{booktabs}
\newcommand{\tabitem}{\llap{}}
\usepackage{adjustbox}
\usepackage{longtable}
\usepackage{footnote}
\makesavenoteenv{tabular}
\makesavenoteenv{table}
\usepackage{footmisc} %reference footnotes
\usepackage{listings}
\usepackage{rotating}
%Put lines numbers in the document
\usepackage{lineno}
\linenumbers{}

\newcommand{\ADD}[1]{\begingroup\color{blue}#1\endgroup}
\begin{document}
\title{A new strategy to characterize the domain architecture structure of 
proteins of the innate inmune system in tunicate species}
% Use \titlerunning{Short Title} for an abbreviated version of
% your contribution title if the original one is too long
\author{Cristian A. Velandia-Huerto*, Ernesto Parra, Federico D. 
Brown, Adriaan Gittenberger, \\ Peter F. Stadler and Clara I. 
Berm\'{u}dez-Santana}


\maketitle

\begin{itemize}
\item \TODO{Include information about D.vexillum sequencing and assembly 
process: Clara and Ernesto.}
\item Adapt draft to journal template.\TODO{Which one?}. 
\end{itemize}

\section*{Introduction}

\PFS{In the last years genomes of non-model organisms have become available
  at a rapidly accelerating rate. As a consequence, their annotation and
  comparative analysis has become a task limited by time and resource
  consumption. Gene architectures serve as a convenient and relatively
  easily accessible source of information about an organism's metabolic and
  regulatory capabilities, and allow the efficient extraction of candidates
  for subsequent, more detailed functional or evolutionary studies
  \cite{aken2016ensembl,birney2004overview,ashburner2000gene,tatusov2000cog,tatusova2016ncbi}.}
  
  \PFS{Conceptually, gene annotation comprises two tasks: first the
  identification of genomic subregions that code for proteins, and second
  the assignment of functionality. Often, both issues are addressed
  simultaneously, using sequence similarity to identify homologs of a query
  with known function and the same time using homology as argument to
  transfer functional annotation. The complexity of eukaryotic genes, with
  its extensive use of alternative splicing and alternative transcription
  starts, however, makes the identification of homologs, and in particular
  the distinction of orthologs and paralogs, a non-trivial, and often
  surprisingly difficult task \cite{yandell2012}. In addition,
  homology-based annotation is by definition limited to know query sets of
  sufficiently well-characterized genes, typically from model organisms.}

  \PFS{Modern gene annotation pipelines therefore are built around
  probabilistic models that are trained on known gene structures. The first
  generation of such tools, such as \texttt{GENSCAN} \cite{genescan}
  primarily focused on promoter signals, intron/exon boundaries, and
  polyadenylation signals \cite{claverie}. State-of-the-art tools, such as
  \texttt{AUGUSTUS} \cite{augustus} or \texttt{GeneID} \cite{Blanco:2007}
  accept diverse types of training data, in particular RNA-seq based
  transcript information. The underlying model can be implemented in very
  different ways: while \texttt{AUGUSTUS} is a generalized Generalized
  Hidden Markov, a rules-based heuristic is used in \texttt{GeneID}. In the
  case of a select few model organisms, gene annotation has evolved into
  major, long-term data curation projects such as VEGA-HAVANA and GENCODE
  for the human genome. \TODO{*** Include a couple of more sentences and a
  more inclusive list of the major curation projects ***} Well curated
  gene models are in important resource also for training gene models for
  application to related species.}
  
  \PFS{The annotation of genes and open reading frames is complemented by
  systematic efforts to establish homology \-\- and in particular orthology
  \-\- information \TODO{ref}. This in turn is forms the basis for defining a
  function-based gene nomenclatures \cite{Braschi:2017}, and efforts to
  achieve a systematic, orthology-aware nomenclature at least across some
  important clades \TODO{** mention VGNC for vertebrated **}
  \cite{aken2016ensembl, birney2004overview}. As a group, vertebrate
  genomes certainly feature the most thoroughly curated and and most
  complete functional annotation.}

In this work we focus on the sub-phylum Tunicata. As sister group of the
vertebrates they occupy a key position in the Tree of Life to understand
the prerequisites for key innovations in the vertebrate lineage. Here, we
are in particular concerned with the evolution of the immune system just
before the ``immunology big-bang'' \cite{bernstein1996} that gave rise to
the origin of the Adaptive Immune System. Like other invertebrates,
Tunicata rely on innate immunity only \cite{franchi2017}. However, they
feature a great diversity of life-styles and the world widely distribution
in ecological niches that may have forced them to evolve different immune
responses to ensure survival in their respective habitats. Since tunicates
can live as solitary sessile or pelagic or to live in colonies they have
complex relationships between the environment, so diversity in the
composition of gene of the immune system is expected
\cite{carroll2008evo,berna2014evolutionary}.

Despite the global importance of this group, genomic studies and
comparative analyses have remained scarce so far. Only the genomes
of three solitary ascidians have been annotated in substantial depth:
the sessiles \textit{Ciona savignyi} and \textit{Ciona intestinalis}
mapped on its $14$ chromosomes \cite{dehal2002draft, small2007haplome} and
the pelagic \textit{Oikopleura dioica}\cite{denoeud2010plasticity,seo2001miniature}. 
And specifically a prediction of miRNAs families were reported for the
solitary \textit{Halocynthia roretzi}\cite{Wang2017} and tree species 
from Molgula sp. (\textit{Molgula occidentalis}, \textit{Molgula oculata} 
and \textit{Molgula occulta})\cite{Stolfi:2014} repored fragmented assemblies
in order to test the idea of developmental system drift along ascidians.
More recently, the genome of colonial ascidians, \textit{Botrillus schlosseri} 
(assembled to $13$ chromosomes) \cite{voskoboynik2013genome} has become available.
The carpet sea squirt \textit{Didemnum vexillum} has been sequenced and
analyzed for its ncRNAs \cite{velandia2016a}. To-date only a very
fragmented draft assembly is available, however. A genome-wide
analisis of the genome sequence from \textit{Botrylloides leachii}\cite{Blanchoud2018} 
reported $1778$ scaffolds and the only one organism of Salpids: 
\textit{Salpa thompsoni}\cite{Obergfell2016} reporting a preliminary
genome assembly that shares along other tunicates high mutation rates.  

Comparisons between tunicate and other chordate genomes have identified
both expansions of gene families but also substantial losses
\TODO{References needed}. The genomic organization of tunicates, as
exemplified by \textit{Ciona} and \textit{Oikopleura} shows substantial
differences compared to both vertebrates and amphioxus, the common outgroup
to the Olfactores \cite{delsuc2006}, and has led different authors to
formulate the idea of the existence in their evolution of processes of
genomic re-structuring in all or some tunicates genomes
\cite{putnam2008amphioxus}.  \TODO{since you say ``different authors'', we
  need several reference here!}  While the other chordate lineages have
maintained a fairly constant rate of evolution, tunicates feature a
systematically accelerated rate of evolution which likely is linked to
specific patterns of organization of their entire gene complement
\cite{putnam2008amphioxus, berna2014evolutionary, Obergfell2016}.

We suspect, therefore, that the chordate immune system also has undergone
substantial changes, restructuring, and diversification. As a first step
towards understanding the evolution of the chordate immune system we
generate her a global overview based on the hypothesis of Paalsson \emph{et
  al.} \cite{paalsson2007building} that the immune system derives from a
small number of ancestral proteins comprising nine ancestral domains.  We
therefore focus in this survey on protein domains as the most elementary
evolutionary building blocks of the immune system and investigate the
turnover of domain architectures as a means to capture at a global level
the evolutionary driving force that led to the complexity of the immune
system in tunicates. In particular, we are interested in the emergence of
novel protein architectures throughout the Tunicata. In order to focus on
gene families that are likely associated with immune system functions, we
consider genes constructed from domains of receptors that are known to be
associated with the innate immune system. It is not uncommon in inmune
system is not rare to find copies and reshuffling of domains
\cite{Forslund2012} \TODO{more references}. We therefore employ here a
domain-based approach to homology search to overcome the limitations of
classical homology search schemes in the face of domain-level changes.

\section*{Theory}

We represent each protein $a$ as an ordered sequence $P(a)$ of domains. In
practice the domains depend on one of several annotation systems, 
\CAVH{but in this study only annotations from \texttt{Pfam} were considered}. Two 
proteins can then be compared a different levels of stringency (\CAVH{after the 
application of the domain reduction, explained in section \textit{Reduction 
System}}):
\begin{itemize}
\item[\textbf{O}rder] \TODO{What exactly is the match criterion?: Do you need
    that every domain of the query
    $Q=(Q_1,Q_2,\dots,Q_n)\in \boldsymbol{\mathfrak{G}}$ matches the
    domain list $ (P_1(a),P_2(a),\dots,P_m(a))$ of a target protein AND that
    the order is preserved? }
    \CAVH{Refers to the comparison of the $Q$ and $P(a)$ domains. It is 
required that order and the domain lists are preserved, too. Repetitions from 
query and subject are represented by one domain.}
\item[\textbf{Disorder}] \TODO{For the writing below you require that the target
    shares two distinct domains with the query list $Q$? Is this correct?
    Or do you mean that you need at least one match for all distinct
    domains and consider only hits with at least two distinct
    domains. Again I could not parse your writing below.} By construction
  every match w.r.t.\ \textbf{O} is also a match w.r.t\ \textbf{D}.
  \CAVH{The Disorder case refers to set comparisons between $Q$ and $P(a)$. 
The number of common elements ($Q \cap P(a)$) must be the same number of 
elements in $Q$ and in $P(a)$. In this case, the order does not be 
preserved.} 
\item[\textbf{Blast}] The domain-based comparisons are complemented by a simple
  \texttt{blast}-based sequence comparison between the query protein
  sequences used to construct $Q$ and the target protein
  sequence. Parameter settings are described in Materials \& Methods.
\item[\textbf{Architecture}] \TODO{I cannot understand how this is different 
from \textbf{O}.} \CAVH{Because Order and Disorder strategies works with 
proteins with $> 2$ domain types (heterodomain proteins), it is required to 
detect those candidates with only one domain family (homodomain proteins). 
Here, was used the implementation by the \texttt{RADS} program  
\cite{Terrapon:2014}. The string of domains comparisons are based on 
an identity matrix score build from Pfam v.30. It is useful to detect not only 
homodomain proteins, but also homology relations between domains with different 
accession numbers.}  
\end{itemize}

The different annotation systems and comparison methods have been integrated 
into a single workflow summarized in Figure~\ref{fig:workflow_golden}.

\begin{figure}[htb]
\begin{center}
\includegraphics[scale=0.17]{figures/completeALLWorkflow3}
\caption{\textbf{A.} Workflow to generate $\boldsymbol{\mathfrak{G}}$. Innate 
immune system databases were used to obtain the accession numbers of the 
related proteins. Next, domain annotation was accesed throught \texttt{biomaRt} 
from \texttt{Ensembl} (v.81). Post processing include reduction of the 
consecutive repetitions of protein domains and finally, the definition of 
$\boldsymbol{\mathfrak{G}}$. \textbf{B.} Methodological steps to obtain innate 
immune system candidates based on $\boldsymbol{\mathfrak{G}}$ definition. 
Used programs from software packages (HHMer and blast) and in-house 
\texttt{Perl} scripts have been highlighted in blue. In green are indicated the 
\texttt{Perl} scripts that perform the reduction function and each step of 
comparison of architectures (A, B, D, O). \TODO{Clean with new and correct definitions.}
}\label{fig:workflow_golden}
\end{center}
\end{figure}

\section*{Methods and Materials}

\subsection*{Comparison Strategies}\label{comparison}

List of gene architecture from tunicates and the other chordates were
greedily \TODO{what does greedily mean here?} compared with elements from
$\boldsymbol{\mathfrak{G}}$ by each $m$ using the followed strategies
\textbf{O}rder, \textbf{D}isorder, \textbf{B}last homology and
\textbf{A}rquitecture or (\textbf{O, D, B, A}) respectively.

\subsection*{Reduction system}\label{reduction}

\TODO{This subsection need complete rewriting. First the Theory section
  needs to be cleaned up.} 

To set out our work, we have defined a reference \textsl{gold standard set}. 
Our survey started building a raw set of domains as follow: let be $G^{a} = 
(P_i,P_{i+1},\ldots,P_{i+k})$ a sub-sequence of ordered domains $P$ in each 
protein $a$ of the innate immune system of organisms taking from 
\texttt{InnateDB} and \texttt{Insect Innate Immunity Database} (IIID) which have 
been annotated by \textit{Pfam} database. Since each domain $P$ has a starting 
$s_k$ and ending $e_k$ point in $a$, we defined an order $P_i \prec P_j$ if and 
only if $s_i \le s_j$. Next, we join all the domains in each protein $a$ as 
\[\bigcup G^{a}\]

Since is very commonly found copies of domains in proteins of the immune 
system, consecutive domains in $G^{a}$ were reduced to a list of unique 
representative domains $P$ if $P_i = P_{i+1}$. From now on we will refer to this 
new set as \textsl{gold standard set} $\boldsymbol{\mathfrak{G}}$ 
(Figure~\ref{fig:workflow_golden}A).
  
\subsection*{Protein domain architectures of reference}
We started with annotated and curated genes from \texttt{InnateDB} 
\cite{Breuer01012013} and \texttt{Insect Innate Immunity Database} (IIID) 
\cite{Brucker2012} in order to define a \textsl{gold standard} set of domain 
architectures of proteins of the innate immune system. At \texttt{InnateDB} many 
other immune-specific databases are linked as \texttt{Immport}, 
\texttt{Immunogenetic related information source (IRIS)}, \texttt{Septic Shock 
Group}, \texttt{MAPK/NFKB Network}, and \texttt{Immunome Database}. Our starting 
point interfaces records from \texttt{InnateDB} to \texttt{Ensembl} (v.86) 
by using \texttt{Perl} scripts and \texttt{biomaRt} R library 
\cite{Durinck:2009aa}. In this step, were mostly retrieved accession numbers 
and sequences belonging to human (GRCh38) and mouse (GRCm38) genomes. Then, to 
increase the set of gene associated with the innate immune system, the 
information from the \texttt{IIID} was used to obtain data of insects like 
\textsl{Nasonia vitripennis}, \textsl{Apis mellifera}, \textsl{Drosophila 
melanogaster}, \textsl{Anopheles gambiae} and \textsl{Acyrthosiphon pisum}. The 
latter genomes were chosen because both have annotations on \textsl{NCBI} and 
\textsl{Ensembl}. For those cases, genes annotated on \texttt{IIID} were retrieved using \texttt{Batch Entrez}\footnote{\url{https://www.ncbi.nlm.nih.gov/sites/batchentrez}}. 

Accession numbers from \texttt{NCBI} were translated into the accession number 
of \texttt{Ensembl}. Then, we proceed to retrieve the data of insects in a 
similar way like in human and mouse. A reference set of domains was obtained 
independently by each domain annotation database after using a \textsl{reduction 
system} described in \textsl{Reduction function subsection}~\ref{reduction}. We 
used the \textsl{gold standard} set for further comparisons of domain 
architectures of $17$ studied species described on Additional File 1.

\subsection*{Re-assembly of \textit{D. vexillum} genome}
\TODO{Include information about re-assembly strategy from D. vexillum, 
methodological steps.}

Annotated ncRNA candidates from the first assembly of \textit{D.\ vexillum} was
mapped in the new assembly as described on Additional File 4. The final genome
coordinates have been reported on \texttt{GFF3} file, including the predicted
gene models by \texttt{AUGUSTUS} and the identified ncRNAs. Additionaly, this
genome annotation could be searched on a genome browser, hosted on
\url{http://tunicatadvexillum.bioinf.uni-leipzig.de/JBrowse-1.16.3/index.html}
\cite{}.

\subsection*{Genomic data sources}

Genomic information source comes from $3$ Vertebrata species:
\textit{Petromizon marinus}, \textit{Danio rerio} and 
\textit{Latimeria chalumnae}, $10$ species of Tunicata: \textit{Oikopleura 
dioica}, \textit{Botryllus schlosseri}, \textit{Botrylloides 
leachii}, \textit{Ciona robusta}, \textit{Ciona savignyi}, \textit{Didemnum 
vexillum}, \textit{Perohora viridis}, \textit{Clavelina oblonga}, 
\textit{Molgula occidentalis} and \textit{Molgula oculata}, $1$ specie 
from Cephalochordata: \textit{Branchiostoma floridae}, represents the final set 
of chosen Chordates. As an outgroup, a set of $2$ species from Echinorderms: 
\textit{Strongylocentrotus purpuratus} and \textit{Patiria miniata} and 
additionally $1$ Hemichordate specie: \textit{Saccoglossus kowalevskii} were 
studied. The protein database sources are described in Additional File 1. 

\TODO{the following is rather incomprehensible. The concepts need to be described in the Theory sect 
see there.} 

\subsubsection*{Architecture Comparison Strategies}
\subsubsection*{\textit{\textbf{O}rder comparison}}

To trace back similar architecture organizations between annotated genes in 
tunicates with the architectures in the \textsl{gold standard set}, tunicate 
domain architectures were represented as query sets  $Q(a) = 
(\mathcal{P}_k,\mathcal{P}_{k+1},\ldots,\mathcal{P}_n)$ and are defined as a 
sub-sequence of ordered domains $\mathcal{P}$ in each protein $a$. Comparing 
the order between $P_i$s and $\mathcal{P}_i$s we defined the number 
$Q(a){success}(o)$

\begin{equation}
  Q(a){success(o)}=\left\{
  \begin{array}{@{}ll@{}}
    1, & \textsl{if}\ P_{i} = \mathcal{P}_{k} \\
    0, & \textsl{otherwise}
  \end{array}\right.
\end{equation} 

If $Q(a){success(o)} = 1$ we say that exist in $\boldsymbol{\mathfrak{G}}$ 
an architecture organization preserving order equal to an architecture 
organization $Q(a)$. If $Q(a){success(o)} = 0$ then we say those 
architectures are not related.

\subsubsection*{\textit{\textbf{D}isorder comparison}}
Since rearrangements of domains are also expected we used a second more 
flexible comparison between elements of $Q(a)$ and 
$\boldsymbol{\mathfrak{G}}$ without considering order in $\mathcal{P}$ domains. 
Now the rules are defined as follows:

\begin{equation}
  Q(a){success(d)}=\left\{
  \begin{array}{@{}ll@{}}
    %1, & Q^a_{m} \subseteq  \boldsymbol{\mathfrak{G}} \quad and \quad 
%\left|Q^a_{m}\right| \geq 2 \\ Re-defined it was wrong written.
    1, & \left|Q^a \cap \boldsymbol{\mathfrak{G}}\right| = 
\left|Q^a\right| = \left|\boldsymbol{\mathfrak{G}}\right| \quad and \quad 
\left|Q^a\right| \geq 2 \\
    0, & \textsl{otherwise}
  \end{array}\right.
\end{equation}

If $Q(a){success(d)} = 1$ we say that exist in $\boldsymbol{\mathfrak{G}}$ 
an architecture composition similar to an architecture organization $Q(a)$. 
If $Q(a){success(d)} = 0$ then we say those architectures are not related. 
Note that here the order of domains  is not a constrain to classify a query set 
$Q(a)$ as success.

\subsubsection*{\textbf{B}last homology comparison}
A classical homology strategy with \texttt{blastp} was used \cite{Korf:2003}. 
For these homology searches, pairwise comparisons were done between the 
proteins used to built both query and \textsl{gold standard} sets. After running 
BLAST following the combination of parameters: 
\begin{lstlisting}[language=bash, breaklines=true]
blastall -p blastp -d <DB> -i <QUERY> -f 9 -F `m S' -M BLOSUM45 -e 100 -b 
10000 -v 10000 -m 8
\end{lstlisting}

were filtered candidate homologous if they satisfied:
\begin{itemize}
\item E-value $\leq 0.001$.
\item Coverage to query length $\geq 60$\%.
\item Identity $\geq 30$\%.
\end{itemize}

\subsubsection*{\textbf{A}rquitecture comparison}
Before the application of reduction system, there are different 
architectures composed by only one domain that had not been taken into account 
with the O,D,B strategies. In order to complement the search strategies, a
comparison between \textsl{gold standard} architectures and query architectures 
was performed applying the methodology reported 
by \texttt{RADS}\footnote{http://domainworld.uni-muenster.de/programs/rads/}\cite{Terrapon:2014}.

First, a domain architecture database was created with the \textsl{gold standard} domains using the program:
\begin{lstlisting}[language=bash, breaklines=true]
makeRadsDB -i <DOMAIN-DISTRIBUTION1> <DOMAIN-DISTRIBUTION2> -s <Seqfasta1> <Seqfasta2> seqFile2.fa -o <OUT-DB>.
\end{lstlisting}

And the comparison was applied against all the query architectures with: 

\begin{lstlisting}[language=bash, breaklines=true]
rads -c -d <DB> -m <Matrix> -q <Input> -o <OUTFILE>
\end{lstlisting} 

Where DB corresponds to the output file from \texttt{makeRadsDB}. Matrix file 
(pfam-30.dsm) was obtained directly from \texttt{RADS} site; Input file 
correspond to the domain's distribution organised as pfam\_scan output 
file. Final candidates were retrieved if reported a similarity normalized score $\geq 0.75$.

\subsection*{Merging output of comparison strategies}
In order to identify the best candidates to be related with the immune 
system, all the previously results from Order, Disorder, Blast and Architecture 
strategies were merged and combined with a \texttt{Perl} script. Candidates 
that have been detected only by Blast (B) strategy were not taken into 
account. Considering all other possible combinations of strategies, it is 
important to note that $O \subset D$, it means that combinations as O, OB, OA, 
OBA are not possible. In this way the remaining combinations $8$ were 
considered to detect the candidates for the innate immune system.

\subsection*{Cleaning specific Hidden Markov Models (HMMs) for each 
domain}
Specific Hidden Markov Models (HMMs) for each domain on the different 
annotation sources were obtained using the program \texttt{hmmfetch} by 
screening on \texttt{Interpro} (Version 60). Then HMMs related with innate 
inmune system in $\boldsymbol{\mathfrak{G}}$ were retrieved. The final list was 
used in further steps.

\subsection*{Screening of architectures domains in Ensembl-non-annotated 
tunicate and cephalochordate species}

\subsubsection*{Ensembl-non-annotated genomes}

The protein annotation for the cephalochordate \textsl{B. floridae} and 
the tunicates \textsl{O. dioica} and \textsl{B. schlosseri} are based on the 
scheme reported at \texttt{JGI Genome Portal} 
(\url{http://genome.jgi.doe.gov/Brafl1/Brafl1.download.ftp.html}) for \textsl{B. 
floridae}, \texttt{Oikoarrays} 
(\url{http://oikoarrays.biology.uiowa.edu/Oiko/Downloads.html}) for \textsl{O. 
dioica} and \texttt{ANISEED} database 
(\url{http://www.aniseed.cnrs.fr/aniseed/download/download_data}) for \textsl{B. 
schlosseri}. In first place, candidates related to the immune system in the 
species \textit{C. robusta}, \textsl{C. savignyi}, \textsl{L. chalumnae}, 
\textsl{P. marinus} and \textit{D. rerio} were used as query sequences to 
perform pairwise homology searches with \texttt{blastp}.

\begin{lstlisting}[language=bash, breaklines=true]
blastall -p blastp -d <DB> -i <QUERY> -F `m S' -m 8 -o <OUT_FILE>
\end{lstlisting}

After filtering, hits candidates with high level of similarity were 
considered as a set of putative candidates, as described below:

\begin{itemize}
\item E-value $\leq 0.001$
\item Coverage $\geq 60$ \%
\item Identity against query $\geq 30$\%.
\end{itemize}

After that, an exhaustive search of HMM domains was conducted by the suite 
\texttt{HMMer} to detect domains using the mapped HMMs in 
$\boldsymbol{\mathfrak{G}}$. Best candidates to annotate protein domains 
derived from \texttt{PFAM} was obtained filtering all of the candidates that 
reported a bitscore $\geq$ Gathering cut-off from \texttt{Pfam} (v.30) and 
reported an internal i-E-value and c-E-value $\leq 0.01$. For the inference of 
domain architecture in these proteins, previously described approach had been 
applied, including the comparison against the \textsl{gold standard} set. The O, 
D, B and A strategies were merged generating the candidates that were 
overlapping between all the strategies, as described on Figure 
~\ref{fig:workflow_golden}.

\subsubsection*{Draft genomes without annotation}

For the recently reported draft genomes of \textit{C.\ oblonga} and 
\textit{P.\ viridis} a \textit{de novo} gene prediction was performed 
directly on the assembled contigs using \texttt{GeneID}\cite{Blanco:2007} 
with the following parameters:

\begin{lstlisting}[language=bash, breaklines=true]
geneid -3 -P <Parameter file> <FASTA FILE> -A >> <GFF3 file>
\end{lstlisting}

Here the \textit{Parameter file} was fetched via \texttt{FTP} for the tunicates: \textsl{C. intestinalis}\footnote{ftp://genome.crg.es/pub/software/geneid/cintestinalis.param\_Apr\_26\_2006} and 
\textsl{O.dioica}\footnote{ftp://genome.crg.es/pub/software/geneid/odioica.param\_Nov\_10\_2006}.
The final result was a \texttt{GFF3} file describing the coordinates on the 
candidate genes, and additionally the set of possible protein candidates in a 
\texttt{fasta} format. Over those candidates \texttt{HMMer} was ran to detect 
domains that intersect with the mapped HMMs in $\boldsymbol{\mathfrak{G}}$. 
Again, filters by each $m$ was used, the \textsl{Reduction 
system}~\ref{reduction} step and comparison strategies \textbf{O, D, B, A} were 
done (Figure~\ref{fig:workflow_golden}).

For the draft genome of the carpet sea squirt \textit{D. vexillum} 
\cite{velandia2016a} a \textit{de novo} gene prediction was performed 
directly on the assembled contigs using \texttt{AUGUSTUS} \cite{augustus} 
with the following parameters:

\TODO{Put AUGUSTUS parameters}

and the complete prediction of homologous architectures was obtained applying 
the pipeline \TODO{Name of my pipeline!}.

%\subsection*{Identification of Nucleotide coordinates and CDS sequences from 
%Protein Domains}
%The nucleotide sequences and the genome coordinates were retrieved for 
%each protein domain that belongs from homologous architectures, inferred from 
%the comparison strategies against \textsl{gold standard}. As shown in Figure 
%~\ref{proteindomains2cds}, the required steps to infer this information for the 
%majority of domains, relies on CDS sequences and annotations at genome level. 
%In 
%case of \textsl{O. dioica}, \textsl{B. schlosseri} and \textsl{B. floridae}, 
%\texttt{GFF} files where downloaded from correspondent databases, including 
%their reported fasta files. In case of \textit{D. vexillum}, the CDS fasta 
%files 
%were inferred directly from the \texttt{GFF} file through a \texttt{Perl} 
%script. For species annotated on \texttt{Ensembl}, transcripts fasta files for 
%each species on their correspondent \texttt{FTP} site\footnote{As described on 
%http://jul2015.archive.ensembl.org/info/data/ftp/index.html} were obtained. 
%With 
%all fasta files from transcripts or CDS, required indexed fasta files were 
%generated with \texttt{makeblastdb}. Fasta sequences from previously 
%identified 
%domains in each candidate proteins were obtained, using \texttt{tblastn} were 
%compared in a pairwise alignment against their correspondent CDS sequences, as 
%described below:

%\begin{lstlisting}[language=bash, breaklines=true]
%tblastn -db <CDS_DB> -query <PROTEIN_DOMAINS_FASTA> -evalue 1000 -word_size 6 
%-window_size 40 -comp_based_stats 2 -gapopen 11 -gapextend 2 -matrix 
%BLOSUM62 -db_gencode 1 -seg no -threshold 21 -outfmt 6 -out <OUT_FILE>
%\end{lstlisting}

%Tabular results were filtered applying the following conditions:

%\begin{itemize}
%\item The name of the possible candidate have to be equal to the 
%previous name of the transcript which it was reported for this protein.
%\item The identity have to be equal to $100$\%. 
%\end{itemize}

%In order to identify the genomic coordinates from these identified CDS, a 
%\texttt{blastn} strategy was applied against the respective reported gene as 
%follows: 

%\begin{lstlisting}[language=bash, breaklines=true]
%blastn -db <Genes DB> <DOMAINS FASTA> -num_threads 8 -evalue 1e^20 -word_size 
%9 
%-gapopen 2 -gapextend 5 -penalty -3 -reward 1 -dust yes -outfmt 6 -out 
%<OUT_FILE>
%\end{lstlisting}

%Specific filters for this methodology were designed (as show below) and the 
%final candidates were organized in a GFF file for each protein database.

%\begin{itemize}
%\item The name of the possible gene candidate have to be equal to 
%the previous name of the gene which it was reported for this CDS or transcript.
%\item The identity have to be equal to $100$\%. 
%\end{itemize}


%\begin{figure}[htbp]
%\begin{center}
%\includegraphics[scale=0.45]{figures/workflowComplete}
%\caption{Methodological steps to obtain innate immune system candidates. 
%Have been highlighted in blue the programs used in this pipeline. In green, 
%those scripts in \texttt{Perl} that performed the reduction function and each 
%of the architecture comparison considered in this study.}
%\label{workflow_domains}
%\end{center}
%\end{figure}

%\begin{figure}[htbp]
%\begin{center}
%\includegraphics[scale=0.3]{figures/workflow_2_dvex}
%\caption{Designed workflow to annotate genes \textit{de novo} in \textit{D. 
%vexillum} genome, and subsequently obtain the architecture distributions 
%according to \textbf{O}rdered, \textbf{D}isorder and \textbf{B}last strategy.}
%\label{workflow_domains_dvex}
%\end{center}
%\end{figure}

%\begin{figure}[htbp]
%\begin{center}
%\includegraphics[scale=0.3]{figures/prot2cds}
%\caption{Designed workflow to annotate genes \textit{de novo} in \textit{D. 
%vexillum} genome, and subsequently obtain the architecture distributions 
%according to \textbf{O}rdered, \textbf{D}isorder and \textbf{B}last strategy.}
%\label{proteindomains2cds}
%\end{center}
%\end{figure}

\subsection*{N-gram analysis on protein domain architectures}
In order to identify the most frequent protein domain along all the protein 
architectures, the final set of strings was analysed as suggested on \cite{Yu:2019}.
In general, the probability of a complete protein, assuming domains as tokens, could be
defined as the chain rule of probability \cite{Jurafsky:2018Book}:
\begin{equation}\label{eqProbAll}
P(d_1,\ldots,d_n) = P(d_1) \times P(d_2|d_1) \times P(d_3|d_1,d_2) \times \cdots \times P(d_n|d_1, \ldots , d_{n-1})
\end{equation}
As noted by \cite{Yu:2019}, before the application of this chain rule, 
it is not a good deal to calculate that using the complete chain of domains, because in this case
the combination of domains is not that fixed and probably the results are scarse. So, 
it is better to consider only the last previous state, and be defined a first order Markov 
process (chains as bigrams). The calculation of the estimated probability was 
performed as suggested in \cite{Yu:2019}:
\begin{equation}\label{eqEST}
P_{MLE} (d_n|d_{n-1}) = \frac{C (d_{n-1},d_n)}{C (d_{n-1})}
\end{equation}
Where, $C(d_{n-1},d_n)$ is the absolute frequency of the bigram and $C(d_{n-1})$ is the number
of bigrams where $d_{n-1}$ is present. So, the way to calculate the probability of 
each domain architecture is given by the product of all relative frequencies of the 
calculated bigrams as described in \cite{Jurafsky:2018Book}, as follows:

\begin{equation}\label{eqFinalProd}
  \begin{split}
    P(w_1^n) \approx \prod_{k=1}^n P(w_k|w_{k-1}) \\ 
  \end{split}
\end{equation}
Where $w_1^n = (w_1, w_2, \dots, w_{n-1})$ is the complete domain architecture, with $n$ domains.

All the bigram probabilities was formated (\texttt{tidyverse}~\cite{wickham:2017}), calculated 
(\texttt{tidytext}~\cite{Wickham:2016}) and plotted (\texttt{igraph}~\cite{Csardi:2006}, 
\texttt{ggraph}~\cite{Pedersen:2018} and ggplot2~\cite{Wickham:2016a}) as a network with 
\texttt{R}.

\subsection*{Orthology detection between candidate innate immune system 
proteins}
In order to detect orthologous groups among innate immune system candidates, 
proteins that reported the same architecture relationships respect to the gold 
standard proteins, were compared using \texttt{ProteinOrtho} (v.5.16) \cite{Lechner2011}, 
as follows:

\begin{lstlisting}[language=bash, breaklines=true]
proteinortho5.pl -force -graph -clean -keep -project=<name-project> 
-step=1 <fasta files>
proteinortho5.pl -force -graph -clean -keep -step=2 
-project={name-project} <fasta files>
proteinortho5.pl -force -graph -clean -keep -step=3  
-project=<name-project> <fasta files>
\end{lstlisting}

Studied species could be grouped into $5$ clades: echinorderms, hemichordates, 
cephalochordates (CE), tunicates (TU) and vertebrates (VE); which have been used 
as a reference to make orthology comparisons. In this case, the species that belong
from echinoderms and hemichordates have been designed as \textsl{Outgroup} (OU) and 
$\boldsymbol{\mathfrak{G}}$ species (GO) have been always considered, in order to create 
orthology comparisons as follows: OU, CE, TU, VE, GO (All species and Golden). Detected 1:1 and 
co-orthologous relationships between pre-defined architecture groups of orthology 
were obtained with a \texttt{Perl} script, as described in \cite{Lechner2011}.

At the same time, available annotation from $\boldsymbol{\mathfrak{G}}$ were obtained from 
\texttt{Ensembl} using \texttt{biomaRt}. For protein candidates that belongs from studied 
species and shared orthologous relations with a $\boldsymbol{\mathfrak{G}}$ protein, the 
retrieved annotation from \texttt{Ensembl} and \texttt{Interpro} accession numbers were 
associated and reported.

\subsection*{Gain and losses of domains}

Reconstruction of family history using Dollo's parsimony was 
achieved with \texttt{Count} \cite{csuros2010}, using the orthology 
results. The presence/absence matrices were obtained using a \texttt{Perl} script 
and the phylogenetic distribution of this species were obtained from 
\cite{Delsuc:2017, Kocot:2018} for tunicates, and for the other organisms from 
Ensembl v.81 compara~\footnote{\url{https://github.com/Ensembl/ensembl-compara/blob/release/81/scripts/pipeline/species_tree.ensembl.topology.nw}}.

\section*{Results}

\subsection*{Global distribution of domains}

A total of $8846$ annotated genes associated with the innate immune system were retrieved 
from \texttt{InnateDB}, of which $7043$ and $1803$ belong to human and mouse 
respectively. After interfaced these records with Ensembl Genome Browser a total 
of $35136$ and $5179$ proteins were identified. Next, to integrate annotations from
the source \texttt{Insect Innate Immunity Database} (IIID) a total of $1312$ proteins were
recovered and were accessed as follows: \textsl{N. vitripennis} $393$ ($368$), 
\textsl{A. mellifera} $170$ ($106$), \textsl{D. melanogaster} $298$ ($242$), \textsl{A. gambiae}
$366$ ($333$) and \textsl{A. pisum} $85$ ($81$), the number in parenthesis corresponds to 
the \textsl{bone fide} annotation in \texttt{Ensembl}. Finally the domain structure was 
traced back with \texttt{biomaRt} in \texttt{Ensembl}. Then, as described in Additional 
File 2, \textsl{gold standard} set ($\boldsymbol{\mathfrak{G}}$) was conformed.

\subsection*{\textbf{ABDO} strategy comparisons of domains}\label{subODB}

The number of genes and proteins that have been included into $\boldsymbol{\mathfrak{G}}$ 
is described in Additional File 2: Table 1. In general, most of the current 
annotations belongs from human ($84.74$\%) and mouse ($12.51$\%) and inside the 
annotation of insects, \textit{N. vitripennis} ($33.14$\%) is the most frequent. 
Those candidates were used as query arquitectures that have been compared through previously 
described \textbf{ABDO} strategies to the subject species (Additional File 1: Table1). 
Taking advantage of the current annotation state is possible to group the query 
species as follows: those ones that have been annotated in \texttt{Ensembl} as
\textit{C.\ robusta}, \textit{C.\ savignyi}, \textit{P.\ marinus}, \textit{D.\ 
rerio} and \textit{L.\ chalumnae}. Also, ones that currently have gene and protein annotations 
but, in independent databases and without the prediction of domains, like \textit{B.\ floridae}, 
\textit{B.\ schlosseri} and \textit{O.\ dioica}. The third group is composed by those 
genomes that have been (re-) assembled in this study, as: \textit{D.\ vexillum}, \textit{C.\ oblonga} and 
\textit{P.\ viridis}. Those species did not have gene or protein predictions, then this 
prediction were performed as described in \textsl{Methods and Materials} with the 
\texttt{GeneID} program using the previously constructed gene models from 
\textit{C.\ robusta} and \textit{O.\ dioica}\footnote{For those genomes, in Table 
~\ref{table:distribution_prot} are referenced as \textsl{Ciro} and \textsl{Oidi} in parenthesis,
respectively} and with \texttt{AUGUSTUS} for \textit{D. vexillum}. Finally, the last group 
is composed by the outgroup species from hemichordates: \textit{S.\ kowalevskii} and from 
echinoderms: \textit{P.\ miniata} and \textit{S.\ purpuratus} and the species from 
tunicates that have now available annotations from protein sequences, as: (\textit{M.\ occidentalis}, 
\textit{M.\ oculata} and \textit{B.\ leachii}), where the \textbf{ABDO} predictions have 
been calculated applying the automated pipeline \TODO{NAME\_PIPELINE} generated in this study.

Current annotation of genes and proteins could reflect the state of assembly
or annotation efforts on subject species. As pointed out in Table~\ref{table:distribution_prot}
is possible to calculate a relationship between the distribution of annotated genes ($g$) and 
their protein-coding products ($p$), as: $p/g$. For \textit{P.\ miniata}, \textit{B.\ floridae}, 
\textit{O. dioica}, \textit{B.\ schlosseri} and \textit{B.\ leachii}, this number reflects a 
1:1 relation, indicating that the annotation for genes are strictly limited for protein-coding genes. 
Larger numbers of reported proteins have been detected for \textit{de novo} predictions of genes, 
where the genomes from \textit{C.\ oblonga} and \textit{P.\ viridis} reported higher number 
of possible protein products. Next, after retrieving or predicting the protein domain annotation, 
always a reduced number of proteins with domain annotation could be identified and particullary for 
the aforementioned genomes with \textit{de novo} predictions, less than $1\%$ of those results had 
a succesfull domain annotation. 

Next with these subset of proteins, domain architectures could be identified, appliying 
the \textsl{Reduction function} described earlier. After independent methods of comparison 
againgst $\boldsymbol{\mathfrak{G}}$, the number of candidates are described for 
each homology architecture strategy (\textbf{ABDO}). The final number of innate 
immune system proteins set is described on column \textbf{Total Prot.\ ISS}, which was 
obtained after merging steps on the ABDO results.

%%%% Discussion about Homology architecture strategies
%%%%%%%%%%
According to the definitions in Material and Methods and Figure~\ref{fig:ABDO}, 
previously defined homology architecture strategies relies on different pairwise 
comparisons, exclusively on protein domain architectures. $\boldsymbol{\mathfrak{G}}$
proteins represent the current annotated proteins on innate immune system and 
their correspondent protein architectures represented as the reduced architectures.
Both, $\boldsymbol{\mathfrak{G}}$ proteins and architectures are used as queries 
in order to perform pairwise comparisons with subject proteins. In this case, 
A (architecture) comparison considers all the annotated protein domains and make a 
comparison based on a precomputed similarity matrix of \TODO{hidden markov profiles} 
in \texttt{RADS} \cite{Terrapon:2014}. B (blast) comparisons does not considers protein 
domains, but the complete sequence of the query and the subject proteins, in this case, 
at least a $70$ \% of similarity has to be considered as a true candidate. D (disorder) 
and O (order) compare the reduced architectures, in this case for the first one, 
all valid combinations of architectures are valid on the subject protein, while the 
number and the identity of the \texttt{Pfam} families are conserved. This is not the case 
of O, where the order of protein domain architecture is an additional requirement to 
consider a homologous subject. 

\begin{figure}
  \begin{tabular}{lcr}
    \begin{minipage}{0.55\textwidth}
    \centering
    \includegraphics[scale=0.4]{figures/ABDO}
    \end{minipage}
    & \qquad &
    \begin{minipage}{0.35\textwidth}
      \caption{Detail of comparisons between query and subject protein domains with 
      different pairwise comparisons: A, B, D, O}\label{fig:ABDO}
    \end{minipage}
  \end{tabular}
\end{figure}


%Following values based on the calculated numbers from: supportFileProteins.xlsx
Differences between strategies were detected when the final numbers of candidates were
 reported. For example, A and B strategies always reported higher frequencies of 
protein candidates in comparison to the results from O or D. Reduced number of
reported candidates are related with the designed filters applied for each
strategy, in this case the main difference relies on the target where the
homology comparison was made. For B, complete protein sequence were taken into
account, A takes the complete arrangement of consecutive collapsed protein domains,
and D or O, uses the collapsed arrangement as a set comparison or a 
list comparison, respectively (See Figure~\ref{fig:ABDO}). Despite the high reports of
possible sequence homology reported with B results, in this approach was not taken
into account unless the relationship was reported in conjunction with another strategy.

In overall results (described on Table~\ref{table:distribution_prot}, last column),
exists a highest distribution from annotated immune system proteins 
in vertebrates (median = $54.48$\% $\pm 7.41$, n=$3$), in comparison to tunicates 
(median = $31.08$\% $\pm 21.58$, n=$12$), cephalochordates ($41.67$\%, n=$1$), and 
the outgroup composed by hemichordates ($56.56$\%, n=$1$) and echinoderms (median = 
$56.105$\% $\pm 0.445$, n=$2$) shows similar distributions. High standard deviation in 
tunicates are a consequence of the inclusion of the new draft genomes 
(from \textit{C.\ oblonga} and \textit{P.\ viridis}), where the prediction of genes 
was \textsl{de novo} by \texttt{GeneID}. In this context, only considering the tunicates 
genomes that had a previous annotation, the estimated values changed 
(median =$38.565$\% $\pm 12.23$, n=$8$) and shown in general, the same reduced values. 
Distinction between colonial (median=$39.66$\% $\pm 15.61$, n=$3$) and 
solitary ($37.47$\% $\pm 9.20$, n=$5$) tunicates does not reported a significant differences
when compared the medians of final reported percentages (Kruskal-Wallis rank sum test,
$p=0.8815, \alpha=0.05$).
%Code executed on: /homes/biertank/cristian/Desktop/Paper_Domains_Figures_Code/StatCalculation/testDifferences.r

%Please copy complete table on K70!
\begin{sidewaystable}
\small
\centering
\begin{tabular}{p{3.2cm}p{2cm}p{2cm}p{2cm}p{2cm}p{2cm}p{2.1cm}p{2.7cm}p{2.6cm}}
\toprule
\textbf{Specie}&\textbf{Annotated Genes}&\textbf{Annotated Proteins}&\textbf{Annotated Prot with domains}&\textbf{Orderded Prot}&\textbf{Disorder Prot}&\textbf{Blast Prot}&\textbf{Architecture}&\textbf{Total Prot IS} \\ 
\midrule
\textsl{P. miniata}&$30399$&$30399$&$20192$ ($66.42$)&$1936$ ($6.37$)&$2161$ ($7.11$)&$11577$ ($38.08$)&$17006$ ($55.94$)&$17151$ ($56.42$)\\
\textsl{S. purpuratus}&$33663$&$35786$&$23640$ ($66.06$)&$3248$ ($9.08$)&$3542$ ($9.90$)&$15420$ ($43.09$)&$19811$ ($55.36$)&$19964$ ($55.79$)\\
\textsl{S. kowalevskii}&$32367$&$22111$&$14888$ ($67.33$)&$1973$ ($8.92$)&$2152$ ($9.73$)&$9737$ ($44.04$)&$12398$ ($56.07$)&$12505$ ($56.56$)\\
\midrule
\textsl{B. floridae}&$50817$&$50817$&$25430$ ($50.04$)&$5499$ ($10.82$)&$4183$ ($8.23$)&$21767$ ($42.83$)&$21000$ ($41.32$)&$21173$ ($41.67$)\\
\midrule
\textsl{O. dioica}&$17212$&$17212$&$5709$ ($33.17$)&$1342$ ($7.80$)&$955$ ($5.55$)&$4577$ ($26.59$)&$4797$ ($27.87$)&$4836$ ($28.10$)\\
\textsl{M. occidentalis}&$30639$&$33023$&$13050$ ($39.52$)&$1195$ ($3.62$)&$1281$ ($3.88$)&$7170$ ($21.71$)&$11192$ ($33.89$)&$11243$ ($34.05$)\\
\textsl{M. oculata}&$15313$&$16616$&$9985$ ($60.09$)&$1336$ ($8.04$)&$1419$ ($8.54$)&$6615$ ($39.81$)&$8472$ ($50.99$)&$8523$ ($51.29$)\\
\textsl{B. schlosseri}&$46519$&$46519$&$8709$ ($18.72$)&$1790$ ($3.85$)&$1264$ ($2.72$)&$6148$ ($13.22$)&$6765$ ($14.54$)&$6847$ ($14.72$)\\

\textsl{B. leachii }&$15839$&$15839$&$9833$ ($62.08$)&$1271$ ($8.02$)&$1422$ ($8.98$)&$6243$ ($39.42$)&$8174$ ($51.61$)&$8284$ ($52.30$)\\
\textsl{C. robusta}&$17153$&$17304$&$12917$ ($74.65$)&$1668$ ($9.64$)&$1160$ ($6.70$)&$6005$ ($34.70$)&$6436$ ($37.19$)&$6484$ ($37.47$)\\
\textsl{C. savignyi}&$12172$&$20157$&$17101$ ($84.84$)&$2087$ ($10.35$)&$1215$ ($6.03$)&$10049$ ($49.85$)&$10079$ ($50.00$)&$10206$ ($50.63$)\\
\textsl{P. viridis} (Ciro)&$6077$&$2221773$&$2806$ ($0.13$)&$56$ ($0.00$)&$61$ ($0.00$)&$12724$ ($0.57$)&$1968$ ($0.09$)&$1972$ ($0.09$)\\
\textsl{P. viridis} (Oidi)&$3025$&$1811030$&$2110$ ($0.12$)&$60$ ($0.00$)&$66$ ($0.00$)&$10329$ ($0.57$)&$1404$ ($0.08$)&$1408$ ($0.08$)\\
\textsl{D. vexillum}&$26546$&$72326$&$36075$ ($49.88$)&$2920$ ($4.04$)&$3654$ ($5.05$)&$16889$ ($23.35$)&$28198$ ($38.99$)&$28686$ ($39.66$)\\
\textsl{C. oblonga} (Ciro)&$19507$&$1174882$&$4032$ ($0.34$)&$120$ ($0.01$)&$125$ ($0.01$)&$4070$ ($0.35$)&$2893$ ($0.25$)&$2902$ ($0.25$)\\
\textsl{C. oblonga} (Oidi)&$4832$&$950470$&$2856$ ($0.30$)&$125$ ($0.01$)&$135$ ($0.01$)&$8746$ ($0.92$)&$1990$ ($0.21$)&$1997$ ($0.21$)\\
\midrule
\textsl{P. marinus}&$13114$&$11444$&$10623$ ($92.83$)&$1650$ ($14.42$)&$1145$ ($10.01$)&$6227$ ($54.41$)&$6025$ ($52.65$)&$6073$ ($53.07$)\\
\textsl{D. rerio}&$31953$&$44489$&$42625$ ($95.81$)&$11762$ ($26.44$)&$4108$ ($9.23$)&$28031$ ($63.01$)&$29523$ ($66.36$)&$29607$ ($66.55$)\\
\textsl{L. chalumnae}&$22628$&$23603$&$22059$ ($93.46$)&$4461$ ($18.90$)&$2185$ ($9.26$)&$9127$ ($38.67$)&$12767$ ($54.09$)&$12858$ ($54.48$)\\
\bottomrule
\end{tabular}
\caption{Final distribution of annotated genes and found candidate Innate
Immune system proteins. The percentage in relation of the total of annotated
proteins (column \textit{Annotated Proteins}) is reported in parenthesis. In
last column \textbf{IIS} refers to: Innate Immune system.}\label{table:distribution_prot}
\end{sidewaystable}

Due to the application of ABDO strategies was independently, it is possible to 
identify the relationships between the query species and the 
$\boldsymbol{\mathfrak{G}}$ species in the final set of immune system 
candidates through the architecture relationships. Different combinations of 
possible architecture comparisons strategies have been merged to four different 
sets, organized by the number of (ABDO) strategies that reported a successfully
architecture comparison. Here, $1 = (A, D)$; $2 = (AB, AD, BD)$; $3 = (ABD, ADO)$ 
and $4 = (ABDO)$; that is $8$ from $15$ possible combinations, because $ (AO, 
BO)$ always map to $ (ADO, BDO)$, due $D \subseteq O$ and additionally, $B$ has 
not been considered at all because this comparison does not represent a pure 
architecture relation, due complete sequence is used in the pairwise alignment 
against $\boldsymbol{\mathfrak{G}}$ protein sequences. Additionaly to this,
the biological relevance and examples for each of the resulted relations are
explained on Table~\ref{table:biologicalexplanation}.

%%%%Biological discussion about homology strategies of architectures
%Total:211849, alone=108368; combined=103481
Based on the final results, the nature of the relationships between subject 
proteins and their correspondent $\boldsymbol{\mathfrak{G}}$ proteins could be explained 
by \textit{one} or \textit{many} ABDO comparisons. For the first case, subject protein
and their correspondent $\boldsymbol{\mathfrak{G}}$ proteins could be well-characterized based
on the description in Table~\ref{table:biologicalexplanation}. Along all results, those
proteins explain at least the $51.15$\% of the relationships, the other group (that represent 
$48.85$\%), are composed by those query proteins that reported more relations and the relations against
$\boldsymbol{\mathfrak{G}}$, could not be easily classified as described in
Table~\ref{table:biologicalexplanation}. 

The relevance to make this distinction is to compare the final number of proteins
that have been detected independently by each ABDO strategies. As shown in Figure~\ref{fig:relation}, 
a relationship between the number of $\boldsymbol{\mathfrak{G}}$ proteins 
and $\boldsymbol{\mathfrak{G}}$ species, was represented in $Y$ axis and then, the complete number
of subject species was used to compare all of the ABDO strategies. Comparisons have been grouped
based on the pre-defined groups of strategies (gray region on the top of plot). In the
group number 1, A strategy recovered all the homodomain proteins with only one annotated domain 
($\sim 78$)\% and the rest are heterodomain proteins with almost similar arrangement of domains,
but without big differences. In comparison to D strategy, A allowed to recognize a higher 
number of proteins and those proteins belong from $\geq 3$ $\boldsymbol{\mathfrak{G}}$ species. D
candidates, are represented by a small subset of heterodomain architectures ($\sim 183$) that
could not be included in A because their domains organization disrupt the overall arrangement
that have been defined for A to be consider as positive. Group $2$ are dominated by AB strategy,
where the arrangement of A has been supported by the sequence homology evaluated by \texttt{blastp}.
In this case, the number of related $\boldsymbol{\mathfrak{G}}$ species and proteins are lower in
comparison to the last group. AD and BD, does not shows too much candidates, as far as the D results 
are few, the possible combinations are possible but not present in more than $3$ 
$\boldsymbol{\mathfrak{G}}$ species. In Group $3$, where all candidates are heterodomain proteins,
ADO and ABD strategies are at some point antagonist, because the strict order 
criteria applied on the first one and not in the second strategy and also, the sequence 
homology is disrupted in the first group. Most of the detected candidates established 
a relationship based on the order and have been detected at the most in $5$ 
$\boldsymbol{\mathfrak{G}}$ species and at the maximum with $120$ proteins. The group 4, combines
all the considered strategies and represents the set of proteins that reported a
conserved arrangement in the domain distribution, both in reduced architecture and the
complete domain annotation. For this group, $\boldsymbol{\mathfrak{G}}$ proteins belong from 
at the maximum from $4$ $\boldsymbol{\mathfrak{G}}$ species and $1$ subject protein 
reported relationships at the most with $35$ $\boldsymbol{\mathfrak{G}}$ proteins.
\TODO{Maybe create another Supplemental files with the plot of proteins and species?
please look into: /homes/biertank/cristian/Desktop/Final\_Figures\_paper\_domains/Code}

%%%%%%%%%%%%%%%%%%%%%%%%
%%% Please explain Fig of relationships and based on values and groups, please
%% detect if exists significative differences. 

\begin{sidewaystable}
\small
\centering
\begin{center}
\begin{tabular}{p{1cm}p{1.2cm}p{9cm}p{10cm}} 
\toprule
\textbf{Group} & \textbf{Strategy} & \textbf{Description}& \textbf{Example}\\
1 & A & Homodomain or heterodomain proteins that reported almost the same arrangement
of protein domains in comparison to $\boldsymbol{\mathfrak{G}}$ proteins. Particulary, 
heterodomains escaped from Order strategy because their protein arrangement does not
follow a strict distribution in query protein, but these variations does not have a big
influence in the final score calculation. At the same time, homodomain proteins reported
 only one domain on their annotations, so also have been not considered into D or O strategies & 
\textbf{PF02460,PF02460} S:Boleac.CG.SB\_v3.S462.g10171.01.p (bole):\textbf{PF02460,PF02460}\\
& D & Proteins that reported the same number and type of domains, but does 
not conserve the arrangement respect to the query $\boldsymbol{\mathfrak{G}}$ architecture. &
Q:ENSP00000358159 (hsa): \textbf{PF00270;PF00271;PF02889} S:Moocci.CG.ELv1\_2.S377548.g10787.01.p (mlis): \textbf{PF02889;PF00270;PF00271} \\
\midrule
2 & AB & Combines both, domain arrangement detected by \texttt{RADS} and local similarity scores 
between conserved regions that are enough to report high similarity using \texttt{blast}. & 
Q:ENSP00000225969 (hsa): \textbf{PF01016} S:ENSDARP00000043545 (dare): \textbf{PF01016} \\
& AD & Proteins that did not reported a strict domain arrangement, but all the same kind of
domains are reported. & Q:ENSP00000284320 (hsa):\textbf{PF00515,PF13181,PF13181,PF13181,PF13181} S:PMI\_006942 (pami): \textbf{PF13181,PF00515} \\
& BD & Proteins that shares the same domains types and also reported local similarities in overall
protein sequence in a pairwise alignment with \texttt{blastp} & Q:ENSP00000481034 (hsa): \textbf{PF07974,PF00008} 
S:ENSCSAVP00000003313 (cisa): \textbf{PF00008,PF00008,PF00008,PF00008,PF00008,PF00008,PF07974,PF00008,PF00008,PF07974,
PF07974,PF00008,PF00008,PF00008,PF00008,PF07974,PF00008,PF00008,PF00008,PF00008,PF00008,PF00008,
PF07974,PF00008,PF00008,PF00008,PF00008,PF00008,PF00008,PF00008,PF07974} \\
\midrule
3 & ABD & Proteins that does not strictly share the same domain arrangement
but the architecture homology is enough to report a true candidates, this 
result is also supported by the pairwise comparison by \texttt{blast} and 
additionally, all the same protein domains conserved between query and subject & 
Q:ENSMUSP00000118471 (mmu): \textbf{PF01421,PF00090,PF05986,PF00090,PF00090}
S:ENSCSAVP00000002867 (cisa): \textbf{PF01421,PF00090,PF05986}\\
& ADO & Proteins that strictly shares the same domain arrangement
and domain families. This group fails in \texttt{blast} searches, so
it means that homology is supported only by pure architecture. & 
Q:ENSP00000233330 (hsa): \textbf{PF08174,PF00169}
S:g15075.t1 (dive): \textbf{PF08174,PF00169}\\ 
\midrule
4 & ABDO & Proteins that reported the most conserved architecture
and sequence homology. Those ones shares the same protein arrangement, 
types of domains and also local homology at pairwise comparisons. &
Q:ENSP00000358159 (hsa): \textbf{PF00270,PF00271,PF02889,PF00270,PF00271,PF02889}
S:GSOIDP00013289001 (oidi): \textbf{PF00270,PF00271,PF02889,PF00270,PF00271,PF02889}\\
\bottomrule
\end{tabular}
\end{center}
\caption{Biological meaning of combination of architecture strategies.}\label{table:biologicalexplanation}
\end{sidewaystable}
%%%%

The use of a defined set of $\boldsymbol{\mathfrak{G}}$ species allowed the detection of
the related candidate proteins on the subject species. As shown in Figure~\ref{fig:FrecEstrat},
it is possible to identify the proportion of shared innate immune system proteins 
respect to $\boldsymbol{\mathfrak{G}}$ species (Figure~\ref{fig:FrecEstrat}). 
As discussed earlier, most of those relationships have been detected with human proteins and
some of them with mouse ones, in general the pattern is conserved along all considered species
in this study, but it could be a consequence of the number of initial candidates on the
$\boldsymbol{\mathfrak{G}}$ proteins, as described on Additional File 2: Table 1.

\begin{figure}[ht!]
	\centering
	\includegraphics[scale=0.7]{figures/relationGoldenABDO}
	\caption{Relationship between number of $\boldsymbol{\mathfrak{G}}$ proteins and
	their correspondent species for each ABDO strategy. Results have been grouped
	according to the number of combined ABDO strategies that detected the relationship
	between query and subject protein.}\label{fig:relation}
\end{figure}


\begin{figure}
  \begin{tabular}{lcr}
    \begin{minipage}{0.6\textwidth}
    \centering
    \includegraphics[scale=0.7]{figures/proportionsGOLD}
    \end{minipage}
    & \qquad &
    \begin{minipage}{0.35\textwidth}
    \caption{Mean shared proportion homology architecture against 
      gold standard species. \textsf{naviP=}\textit{N.\ vitripennis}, 
      \textsf{apmeP=}\textit{A.\ mellifera}, \textsf{drmeP=}\textit{D.\ melanogaster}, 
      \textsf{angaP=}\textit{A.\ gambiae} and \textsf{acpiP=}\textit{A.\ pisum}; 
      and Mammals: \textsf{mmuP=}\textit{M.\ musculus} and 
    \textsf{hsaP=}\textit{H.\ sapiens}.}\label{fig:FrecEstrat}
    \end{minipage}
  \end{tabular}
\end{figure}

\subsection*{Relationships between Innate Immune system candidates}\label{Orthology}

Current detected innate immune system candidates on subject species represent
a set of annotated proteins that could be subject of better annotations or
classifications based on calculated information given by this study. By this
way, the identity of the protein domain profiles could be taken into account
and be described based on their domain profile composition: homodomain 
(by the same type of domains) or heterodomains (more than one domain type)
(Figure~\ref{fig:domainDistr}). By practical means all the species was grouped 
in $5$ taxonomical clades as: \textbf{Echi} = Echinodermata, \textbf{Hemi} = 
Hemichordata, \textbf{Ceph} = Cephalochordata, \textbf{Tuni} = Tunicata and 
\textbf{Vert} = Vertebrata, following the sub-phylum assignment on Additional 
File 1:Table 1. Homodomain proteins are depicted on Figure~\ref{fig:domainDistr}A, 
where a total number of $2375$ homodomain architectures were detected along all 
species: echinoderms ($77.28$\%), hemichordates ($72.56$\%), cephalochordates 
($64.04$\%), tunicates ($44.58$\%) and vertebrates ($65$\%). This distribution 
also shows a high variation in total number of architectures of one type of domain
in echinoderms and tunicates, respect to vertebrates.

\begin{figure}[ht!]
\centering
\includegraphics[scale=0.53]{figures/completeDistributionDomains} \\
%\includegraphics[scale=0.38]{figures/heterodomains_distr}
\caption{\textbf{A} Homodomain architecture distribution. 
	\textbf{B} Multidomain architecture distribution. \textbf{Echi:} 
	Echinodermata, \textbf{Hemi:} Hemichordata, \textbf{Ceph:} Cephalochordata, 
	\textbf{Tuni:} Tunicata and \textbf{Vert:} Vertebrata.
}\label{fig:domainDistr}
\end{figure}

In the homodomain set, the most frequent protein architecture with 
current annotation from \texttt{Pfam} is Rhodopsin-like receptor 
(PF00001) for echinoderms, hemichordates and vertebrates, while for 
Cephalochordata was LRR\_8 (PF13855) and for tunicates the Protein 
Kinase domain (PF00069). 

 
%%%HOMO
%  Clade        SD     Mean  Min  Max  N Total Est Perc<Total>
%1  Ceph        NA 1521.000 1521 1521  1  1521 1521 0.6404
%2 Echin  98.28784 1835.500 1766 1905  2  3671 1835.5 0.7728
%3  Hemi        NA 1721.000 1721 1721  1  1721 1721 0.7256
%4  Tuni 423.51523 1058.667  467 1727 12 12704 1058.7 0.4458
%5  Vert 362.95225 1543.667 1213 1932  3  4631 1543.7 0.65

%ORDER
% PF00069  Tuni  2326
% PF00001  Vert  1307
% PF00001 Echin  1182
% PF13855  Ceph  1049
% PF00001  Hemi   477

%%
%Despite the similarity of this results along all clades, distribution 
%from \textsl{B.\ floridae} (Cephalochordata) showed also higher number of 
%proteins with $2$ and $3$ types of domains with similar numbers as reported 
%on vertebrates; but not in echinoderms, hemichordates and tunicates, that 
%showed similar density distributions. Also, few proteins are composed by more 
%than $5$ different types of domains, but the current distributions shows, in 
%all clades, a very long heavy tails with \TODO{$\ge 20$ domains types}.
%%

%%Hetero
%  Clade    SD  Mean   Min   Max     N Total Est Perc
%  <fct> <dbl> <dbl> <dbl> <dbl> <int> <int> <1S> <Total>
%1 Ceph   NA    937    937   937     1   937 937 0.464
%2 Echin  90.5  834    770   898     2  1668 834 0.413
%3 Hemi   NA    861    861   861     1   861 861 0.426
%4 Tuni  263.   408     46   717    12  4896 408 0.202
%5 Vert  434.  1072.   638  1505     3  3217 1072.3 0.531
%%%

% Vert 431 PF13912,PF13913 C2H2-type zinc finger,zinc-finger of a C2HC-type
% Tuni 556 PF00651,PF07707,PF01344,PF13964 BTB/POZ domain,BTB And C-terminal Kelch,Kelch motif,Kelch_6
% Ceph 269 PF00651,PF07707,PF01344,PF13964 BTB/POZ domain,BTB And C-terminal Kelch,Kelch motif,Kelch_6
% Hemi 116 PF02931,PF02932 Neur_chan_LBD,Neur_chan_memb
% Echin 181 PF15227,PF00643 zf-C3HC4_4,B-box zinc finger

%%%%
%Please, all the results after bigram model are in: ''~/Desktop/Final_Figures_paper_domains/Code/calculate_estimation_probability.sh``
%And results are the table: table.txt and the plot distr_bigrams.pdf
A total of $2020$ heterodomain architectures have been detected along all
the set of proteins. Specifically, for all clades, was possible to calculate 
the found average percentage of architectures respect to the total number of 
heterodomain architectures for all clades: echinoderms ($41.3$\%), 
hemichordates ($42.6$\%), cephalochordates ($46.4$\%), tunicates 
($20.2$\%) and vertebrates ($53.1$\%). 

As a complementary analysis, an n-gram analisis was performed on the complete set of 
IIS domain architectures and their associated probability was calculated. 
As suggested \cite{Yu:2019} n-gram analyses could be used to calculate the 
relative frequency of bigrams and also the probability over final set of architectures.

\begin{figure}[ht!]
  \centering
  \includegraphics[scale=0.2]{figures/bigramsFinal}
  %\includegraphics[scale=0.3]{figures/network_10_NO_bigrams}
  \caption{Bi-grams network. Only was considered those bigrams that
  have been reported $\geq 10$ along all the architectures.
  X corresponds to N-terminus and Z to the C-terminus.}\label{fig:bigramNet}
\end{figure}

In accordance to the Materials and Methods, bigrams have been derivated from the
n-gram analisis and as an intermediate results, absolute and relative frequencies
were calculated. With this information, was possible to identify the most 
frequent bigrams, including the N-terminus and the C-terminus positions along all 
architectures. Figure~\ref{fig:bigramNet} shows the network of the most
frequent bigrams, and the direction of the vertices represents the bigram 
relationship between domains ($w_1 \rightarrow w_2$). In more detail,
Figure~\ref{fig:bigramNet}A both, N and C-terminus are central nodes 
but with opposite order relations, as expected due the arrangement inside the bigram. 
When both terminals are not considered into the graph, only relationships of 
domains have been considered (Figure~\ref{fig:bigramNet}B). In this case, 
$8$ clusters of bigrams have been detected and the domain associated annotation 
could be retrieved from \texttt{Pfam}. In this case as shown in 
Table~\ref{tab:meansDomains}, domains are related with cell surface receptors, 
coagulation, cell-adhesion, membrane-bound proteins and calcium dependent 
proteins (Group 1). Group 2 is composed by various plasma proteins and cysteine-rich 
domains. Group 3 are characterized by Zinc finger containinig proteins
that tipically takes part of molecular scaffolds between protein and other molecules\footnote{
\url{https://www.ebi.ac.uk/interpro/potm/2007_3/Page1.htm}}. Inside Group 4, a conserved
transduction domain as RhoGEF\footnote{\url{http://www.ebi.ac.uk/interpro/entry/IPR000219}} 
and recruiting domain PH\footnote{\url{http://www.ebi.ac.uk/interpro/entry/IPR001849}}
are related also in the signal transduction pathways. Group 5 is composed by the conserved
Fibronectin domains III (FN3), that is commonly identified on cell-surface receptors 
and extracellular proteins\footnote{\url{http://www.ebi.ac.uk/interpro/entry/IPR003961}} 
and also, a complete set of Ig domains, which are related in recognition, binding and 
adhesion processes, particulary the folding of this domains is similar to FN3, cadherin and 
cytokine domains, but differs in their functions \cite{Barclay:2003}. Group 6 are
composed by repeated recognition leucine rich patterns (LRRs) that have been 
detected on different species and are recognized as key domains in protein-protein 
interactions or recognition\footnote{\url{https://pfam.xfam.org/family/PF13855}}. 
Direct interactions with DNA are connected with the folding task of DEAD box helicales, 
which are related to RNA metabolism processes\footnote{\url{http://www.ebi.ac.uk/interpro/entry/IPR011545}}
(Group 7). Finally, Group 8 are composed by the conserved Ankyrin repeats that have
been classified as one of the most common protein-protein interaction 
domains\footnote{\url{http://www.ebi.ac.uk/interpro/entry/IPR002110}}. 
It is important to point out that protein domains have been classified in terms of 
bigrams and their absolute distribution along all dectected combinations, but this
frequency adjacencies/relations could be supported by their biological functions, where
most of the protein domains share functions taking part on recognition, 
transduction or modification taks inside/outside the cell.

\begin{table}
\centering
\begin{tabular}{p{1cm}p{1.5cm}p{7cm}p{5cm}}
  \toprule 
  \textbf{Group} & \textbf{Numer of domains} & \textbf{Central Domain} & \textbf{Other domains} \\
  \midrule 
  1 & 6 & PF07645 (Calcium-binding EGF domain)& PF00058 (Low-density lipoprotein receptor repeat class B),
  PF14670 (Coagulation Factor Xa inhibitory site), PF00683 (TB domain), PF00008 (EGF-like domain), 
  PF02210 (Laminin G domain) \\
  2 & 4 & PF01826 (Trypsin Inhibitor like cysteine rich domain), PF00094 (von Willebrand factor type D domain), 
  PF08742 (C8 domain) & PF00090 (Thrombospondin type 1 domain)\\
  3 & 2 & PF13912 (C2H2-type zinc finger), PF00096 (Zinc finger, C2H2 type) & NA \\
  4 & 2 & PF00169 (Pleckstrin homology domain), PF00621 (RhoGEF domain) & NA \\
  5 & 4 & PF00041 (Fibronectin type III domain), PF07679 (Immunoglobulin I-set domain), 
  PF13927 (Immunoglobulin domain Ig 3), PF13895 (Immunoglobulin domain Ig 2) & NA \\
  6 & 3 & PF01462 (Leucine rich repeat N-terminal domain), PF13855 (Leucine rich repeat),
  PF01463 (Leucine rich repeat C-terminal domain) & NA \\
  7 & 2 & PF00270 (DEAD/DEAH box helicase), PF00271 (Helicase conserved C-terminal domain) & NA \\
  8 & 2 & PF00023 (Ankyrin repeat), PF12796 (Ankyrin repeats (3 copies)) & NA \\
  \bottomrule 
\end{tabular}
\caption{Annnotation of bigram groups and network clusters. \TODO{Maybe on Supp info.}}\label{tab:meansDomains}
\end{table}

As shown in Figure~\ref{fig:probArch}, the distribution of the probability of the 
architectures in both, homodomain and heterodomain arrangements reported an unimodal 
and bimodal density, respectively. Both, have long tails which corresponds to architectures that
are the most probables along the complete set of architectures related with the IIS, 
specifically the distribution of homodomains are the most probable architectures 
(i.e. PF00069 (Protein kinase domain), PF00169 (PH domain), PF00595 (PDZ domain), 
PF00076 (RRM\_1 or RNA recognition motif) and PF00400 (WD domain)). In average, around $3$ times
more probable than the most probable heterodomain proteins. By the same way, inside 
heterodomains the $5$ most probable architectures are: PF08953 (DUF1899)-PF00400 (WD domain); 
PF13855 (Leucine rich repeat)-PF01582 (TIR domain); PF00270 (DEAD/DEAH box helicase)-PF00271 
(Helicase conserved C-terminal domain); PF13912 (C2H2-type zinc finger)-PF00096
(Zinc finger, C2H2 type) and PF13927 (Ig\_3)-PF07679(Immunoglobulin I-set domain)
(for complete list see Supplemental File X), along those architectures, the majority of 
are composed by $2$ domains. \TODO{Please create Supplemental File with all table}.

%> summary(hom$Prob)
%     Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
%5.688e-05 2.275e-04 2.275e-04 2.382e-04 2.275e-04 2.151e-03 
%> summary(het$Prob)
%     Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
%0.000e+00 2.855e-05 1.011e-04 1.166e-04 2.184e-04 6.027e-04

\begin{figure}
  \begin{tabular}{lcr}
    \begin{minipage}{0.6\textwidth}
    \centering
    \includegraphics[scale=0.6]{figures/distr_bigrams_complete} 
    \end{minipage}
    & \qquad &
    \begin{minipage}{0.35\textwidth}
    \caption{Density of probabilities of final hetero/homodomain architectures related to the
  innate immune system.}\label{fig:probArch}
    \end{minipage}
  \end{tabular}
\end{figure}

\TODO{When the architecture distribution is considered, the most frequent along 
	all species are described on Addtional file. Change this table 
	describing by specie, clade or even describe the matrix based on 1:1 
	orthologous proteins.
}


%Protein clusters made by the commmon protein architecture
Once protein candidates were detected by ABDO strategies on studied species, 
a further step is the identification of the biological relevance of the 
new detected protein candidates. Applying a clustering strategy 
based on sharing the same group of $\boldsymbol{\mathfrak{G}}$ proteins,
allow to create $4787$ groups to access for orthology relationships with 
described methodology through \texttt{ProteinOrtho}. After retriving the
results the first step to study those relations is the identification of 
orthologous (1:1) or co-orthologous (1:many or many:many) groups. 
A total of $30782$ relations were detected, from them: $52.55$\% reported 1:1 
relationships, while co-orthologs are represented by $47.45$\%. Inside 1:1 relations, 
exists $3726$ that includes at least one $\boldsymbol{\mathfrak{G}}$ protein 
and most of them are relations involved $2$ species ($52.20$\%) and 
the distribution reach a maximum of $15$ species but it was detected only 
in $1$ orthology group. 

%Analize results from Trees!

Then, in order to identify the phylogenetically distribution of those 1:1 
proteins, a presence/absence matrix was calculated and phylogenetic tree 
of studied species was obtained from \cite{} to calculate evolutionary history 
of orthologous proteins from IIS by the Dollo parsimony \cite{}.

As represented on Figure~\ref{fig:dollooneone}, a set of $1593$ architectures 
are shared in the base of the \TODO{Deuterostomia?} and in both clades: 
chordates and ambulacrarians report only gains (g) events. In this way, at the 
base of Chordata could be identified $g=109$ and in comparison to \textit{Ambulacraria}
the pattern was the same but with fewer gains $g=36$. In total \texttt{Chordata}
reported $1702$ present domain architectures, which have been lost about
$l=843$ in \textit{Cepahlochordata} without any gain events. In comparison
to \textit{Olfactores}, this group reported few losses and a clade-specific gains 
(l=$34$, g=$262$). In the divergence between \textit{Tunicata} and 
\textit{Vertebrata}, both clades reported specific gains and losses, but with a 
higher number of losses that could be traced in the base of vertebrates 
($l=409$, $g=35$). Gain and loss in tunicates could be traced ($g=16$,$l=181$), but not with
the same magnitude as occurred in vertebrates, reporting at the end $1765$ architectures in
the base of the clade. More into details, about $73.94$\% of those architectures 
have been lost in \textit{O. dioica} (Appendicularians) and also reporting a very 
few number of gains ($g=11$). \TODO{The clade that groups} \textsl{Stolidobranchia, 
Phlebobranchia and Aplousobranchia} increased the total number of architectures. 
In comparison, more loss events have been detected in the clade (Phlebobranchia +
Aplousobranchia) ($g=13$, $l=368$) than Stolidobranchia ($g=30$, $l=308$). In 
overall through those clades is important to note that Aplousobranchia reported 
between almost $1.7\times$ loss events ($g=3$, $l=531$) than Phlebobranchia ($g=2$, $l=307$) 
and Stolidobranchia ($g=30$, $l=308$). At the same time, exists a high number of lost 
architectures in species where \textit{de novo} gene prediction was predicted (with 
gene models from \textit{C.\ robusta} and \textit{O.\ dioica} using \texttt{GeneID}). 

%Here calculating the average of specie-specific architectures based on the
%terminal numbers
In terms of numbers, Stolidobranchia reported in average, biggest value of 
architectures ($924.5 \pm 153.61$, $n=4$) than Aplousobranchia 
($518.3 \pm 244.62$, $n=3$) and Phlebobranchia ($547.5 \pm 272.96$, $n=4$).
And inside tunicates, \textit{D.\ vexillum} reported the highest number of
gains ($g=24$, $l=163$). At the same time, inside vertebrates the 
highest number of architectures are present on \textit{D.\ rerio}, 
not only the specie-specific gain events ($g=152$, $l=185$), but also 
gains at the base of Osteichthyes ($g=122$, $l=131$). 

\begin{figure}[ht!]
\centering 
\includegraphics[scale=0.63]{figures/oneoneDollo} \\
\caption{Evolutionary history of 1:1 orthologous protein architectures in 
Deuterostomata. \TODO{This is a temporal image, because I have to plot better this tree}}\label{fig:dollooneone}
\end{figure}

\texttt{Interpro} annotations was retrieved with \texttt{biomaRt} for 
those $\boldsymbol{\mathfrak{G}}$ proteins that have been identified
as 1:1 orthologs of IIS candidates on studied species.
\TODO{How visualize those results?}

\TODO{Final tables are reported in Additional Files 3 
(oneone\_heterodomains\_annotation\_proteins.txt, 
oneone\_homodomains\_annotation\_proteins.txt)}

\TODO{TODO:}
\begin{itemize}
\item \TODO{In this case, I have the interpro annotation for a given architecture based
on the golden protein ($\boldsymbol{\mathfrak{G}}$) which have a 1:1 relationship. Just
reporting those data? or is more convenient to plot by some way?}
\item \TODO{Because the paper is about innate immune system, I was looking for 
categories which I could classify the final architectures. I have found a nice classification
specifically for tunicates on \url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5465252/pdf/fimmu-08-00674.pdf} but I have to take some examples for each category and look it into
the found candidates. Another option is on this systematic revision of domains: \url{https://www.sciencedirect.com/science/article/pii/S0378111918311119?via\%3Dihub}.}.
\item \TODO{Dollo parsimony could be applied for analize gain and loss of protein architectures?.}
\item \TODO{I could represent the protein architectures as drawings of the proteins, or a comparison between species, but for the most important ones in innate immune system, there is lot of information.}
\item \TODO{I would like to report the ABDO method. I have the program, mainly Perl scripts and glue code in bash or maybe in a server?}
\end{itemize}

\section*{Conclusions}
\bibliographystyle{abbrvnat}
\bibliography{biblio,otherbibio}

\end{document}
